resolver.go

```diff
@@ -388,7 +391,7 @@ func (r *resolver) ServeDNS(w dns.ResponseWriter, query *dns.Msg) {
 
 			// forwardQueryStart stores required context to mux multiple client queries over
 			// one connection; and limits the number of outstanding concurrent queries.
-			if r.forwardQueryStart(w, query) == false {
+			if r.forwardQueryStart(w, query, queryID) == false {
```

Comment:
forwardQueryStart() changes the ID in the query to the one allocated by embedded Server. After sending the query to a first resolver if it timesout or gets a response but couldn't retrieve the context we continue and send it to the next resolver. When doing that the ID we have to save is the one that we got from the client and not the one in the current query because its the changed value.